litestar/_signature/utils.py

```diff
@@ -110,11 +110,7 @@ def _get_signature_model_type(
     preferred_validation_backend: Literal["pydantic", "attrs"],
     parsed_signature: ParsedSignature,
 ) -> type[SignatureModel]:
-    if preferred_validation_backend == "attrs" or _any_attrs_annotation(parsed_signature):
-        from litestar._signature.models.attrs_signature_model import AttrsSignatureModel
-
-        return AttrsSignatureModel
-    return PydanticSignatureModel
+    return MsgspecSignatureModel
```

Comment:
We no longer need the abstraction layer which makes signature models generic. It can be removed..

Reply:
absolutely - there is a lot to be tidied up before this PR is ready - just shooting for MVP at the moment